src/data_generators/synthetic_data_generator.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2021 The Private Cardinality Estimation Framework Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Generates a DataDesign from DataDesignParameters."""
+
+from absl import app
+from absl import flags
+import math
+from typing import List
+from wfa_planning_evaluation_framework.data_generators.data_design import DataDesign
+from wfa_planning_evaluation_framework.data_generators.data_set import DataSet
+from wfa_planning_evaluation_framework.data_generators.data_set_parameters import (
+    DataSetParameters, GeneratorParameters)
+from wfa_planning_evaluation_framework.data_generators.pricing_generator import PricingGenerator
+from wfa_planning_evaluation_framework.data_generators.fixed_price_generator import FixedPriceGenerator
+from wfa_planning_evaluation_framework.data_generators.impression_generator import ImpressionGenerator
+from wfa_planning_evaluation_framework.data_generators.homogeneous_impression_generator import HomogeneousImpressionGenerator
+from wfa_planning_evaluation_framework.data_generators.publisher_data import PublisherData
+from wfa_planning_evaluation_framework.data_generators.test_synthetic_data_design_config import TestSyntheticDataDesignConfig
+from wfa_planning_evaluation_framework.data_generators.synthetic_data_design_config import SyntheticDataDesignConfig
+from numpy.random import RandomState
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('output_folder', 'TestDataDesign', 'Output Folder.')
+flags.DEFINE_string('data_design_config', 'TestConfig', 'Data Desgin Config.')
+
+name_to_config_dict = {'test': TestSyntheticDataDesignConfig}
+
+
+class SyntheticDataGenerator():
```

Comment:
One thing that I think might be an improvement that we didn't discuss previously would be to create a single np.random.Generator object in the __init__ function, perhaps optionally seeded by a value given on the command line, and then to pass this generator to all of the objects that need a random state.  This would help to eliminate any possible strange behavior due to different objects using the same random seed, or due to overlook a random seed somewhere.  WDYT?

Reply:
asking offline..

Reply:
sounds great, I have some questions about this, asking offline..

Reply:
Updated based on our discussion today. However, I ended up using RandomState rather than Generator. This is because we are dependent upon the cardinality eval framework and that uses RandomState. In order to migrate to Generator, we would need to do a overall refactor. I think it is not critical and we can pick that up later if necessary. wdyt?

Reply:
I don't see the need for an overall refactor.  Can you explain?  If a function from the cardinality eval framework requires a random_state object, it seems to me that we can generate one on the fly using the expression np.random.RandomState(seed = g.random()), where g is the Generator.  I would prefer that the Planning Eval Framework use Generators throughout, except where passing RandomState parameters to legacy code (e.g., cardinality eval fw), which can be handled as above.

Reply:
sounds good, thanks for the explanation.